### PR TITLE
Port more CV tests

### PR DESCRIPTION
**Pull request by oshtaier**
to avoid regression:
```
py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contenthost.py
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-06 17:15:47 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui_airgun/test_contenthost.py::test_positive_end_to_end PASSED                                                                                                           [100%]

================================================================================= 1 passed in 688.33 seconds ===============================================


py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-06 16:55:52 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui_airgun/test_contentview.py::test_positive_end_to_end PASSED                                                                                                           [100%]

================================================================================= 1 passed in 187.79 seconds ============================================================================


py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_hostcollection.py::test_positive_install_errata
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-06 17:30:57 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui_airgun/test_hostcollection.py::test_positive_install_errata PASSED                                                                                                    [100%]

================================================================================= 1 passed in 814.48 seconds ===========================================================================

```
new tests:
```
py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_search_composite
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: ../robottelo/.pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-08 13:45:43 - conftest - DEBUG - BZ deselect is disabled in settings


../robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_search_composite PASSED                                                                                        [100%]

================================================================================= 1 passed in 86.72 seconds ======================================================================


py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_publish_with_force_puppet_env
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: ../robottelo/.pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-08 13:58:38 - conftest - DEBUG - BZ deselect is disabled in settings


../robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_publish_with_force_puppet_env PASSED                                                                           [100%]

================================================================================= 1 passed in 360.50 seconds ================================================================

py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_publish_with_repo_with_disabled_http
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: ../robottelo/.pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-08 14:05:12 - conftest - DEBUG - BZ deselect is disabled in settings


../robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_publish_with_repo_with_disabled_http PASSED                                                                    [100%]

================================================================================= 1 passed in 259.94 seconds ==================================================================

py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_publish_promote_with_custom_puppet_module
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: ../robottelo/.pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-08 14:10:16 - conftest - DEBUG - BZ deselect is disabled in settings


../robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_publish_promote_with_custom_puppet_module PASSED                                                               [100%]

================================================================================= 1 passed in 291.83 seconds =======================================================================


py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_delete_with_kickstart_repo_and_host_group
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: ../robottelo/.pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-08 14:16:57 - conftest - DEBUG - BZ deselect is disabled in settings


../robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_delete_with_kickstart_repo_and_host_group PASSED                                                               [100%]

================================================================================= 1 passed in 456.38 seconds ===============================================================================


py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_custom_ostree_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: ../robottelo/.pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-08 14:30:21 - conftest - DEBUG - BZ deselect is disabled in settings


../robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_custom_ostree_end_to_end PASSED                                                                                [100%]

================================================================================= 1 passed in 373.59 seconds ========================================================


py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_rh_ostree_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: ../robottelo/.pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-08 14:43:21 - conftest - DEBUG - BZ deselect is disabled in settings


../robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_rh_ostree_end_to_end PASSED                                                                                    [100%]

================================================================================= 1 passed in 653.50 seconds ============================================================

py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_mixed_content_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: ../robottelo/.pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-08 14:54:50 - conftest - DEBUG - BZ deselect is disabled in settings


../robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_mixed_content_end_to_end PASSED                                                                                [100%]

================================================================================= 1 passed in 495.72 seconds ===================================================================


py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_rh_mixed_content_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: ../robottelo/.pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-08 15:03:28 - conftest - DEBUG - BZ deselect is disabled in settings


../robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_rh_mixed_content_end_to_end PASSED                                                                             [100%]

================================================================================= 1 passed in 568.05 seconds =====================================================
```

### CODE REVIEW

File: tests/foreman/ui/test_contentview.py [hunk 1]
```diff
@@ -2675,915 +2526,37 @@ def test_positive_delete_version_without_refresh(self):
...
-        env_name = gen_string('alpha')
-        cv_name = gen_string('alpha')
-        with Session(self, username, user_password) as session:
-            # Create Life-cycle environment
-            make_lifecycle_environment(session, name=env_name)
-            # Creates a CV along with product and sync'ed repository
-            self.setup_to_create_cv(repo_name=repo_name)
-            # Create content-view
-            make_contentview(session, name=cv_name)
+            make_contentview(session, name=cv_name)
```
Comment (abalakh) [inline tests/foreman/ui/test_contentview.py]: Is it an issue with diff or smth was really added to this file?

File: tests/foreman/ui_airgun/test_contentview.py [hunk 2]
```diff
@@ -1350,3 +1362,503 @@ def test_positive_update_filter_affected_repos(session, module_org):
...
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+    """
+    composite_name = gen_string('alpha')
+    with session:
+        session.contentview.create({
+            'name': composite_name, 'composite_view': True})
+        assert session.contentview.search('composite = true')[0]['Name'] == composite_name
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_contentview.py]: Kinda risky - you're reusing existing org where some other composite cv may be found. Maybe try `composite_name in {ccv['Name'] for ccv in session.contentview.search('composite = true')}` instead?
Or query containing both name & composite.

File: tests/foreman/ui_airgun/test_contentview.py [hunk 3]
```diff
@@ -1350,3 +1362,503 @@ def test_positive_update_filter_affected_repos(session, module_org):
...
+    # Creates new custom repository via API's
+    repo = entities.Repository(
+        url=FAKE_0_PUPPET_REPO,
+        content_type=REPO_TYPE['puppet'],
+        product=product,
+    ).create()
+    # Sync repo
+    call_entity_method_with_timeout(
+        entities.Repository(id=repo.id).sync, timeout=1500)
+
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_contentview.py]: nitpick: redundant empty line

File: tests/foreman/ui_airgun/test_contentview.py [hunk 4]
```diff
@@ -1350,3 +1362,503 @@ def test_positive_update_filter_affected_repos(session, module_org):
...
+    # Create a new Lifecycle environment
+    lc_env = entities.LifecycleEnvironment(organization=org).create()
+
+    # Create a Product and Kickstart Repository for OS distribution content
+    product = entities.Product(organization=org).create()
+    repo = entities.Repository(product=product, url=settings.rhel6_os).create()
+
+    # Repo sync procedure
+    call_entity_method_with_timeout(repo.sync, timeout=3600)
+
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_contentview.py]: nitpick: imo these empty lines are redundant as well - your test contains nice comments which help to distinguish commands not worse than empty lines.

File: tests/foreman/ui_airgun/test_contentview.py [hunk 5]
```diff
@@ -1350,3 +1362,503 @@ def test_positive_update_filter_affected_repos(session, module_org):
...
+    with session:
+        # Create new content view
+        session.contentview.create({'name': cv_name})
+        assert session.contentview.search(cv_name)[0]['Name'] == cv_name
+
+        # Add OSTree repository to content view
+        session.contentview.add_ostree_repo(cv_name, repo_name)
+        cv = session.contentview.read(cv_name)
+        assert cv['ostree_content']['resources']['assigned'][0]['Name'] == repo_name
+
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_contentview.py]: same here

**Comment by oshtaier** [comment]
```
py.test -v /home/ashtayer/Desktop/robottelo/tests/foreman/ui_airgun/test_contentview.py::test_positive_search_composite
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collected 1 item                                                                                                                                                                             
2018-11-14 16:22:46 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui_airgun/test_contentview.py::test_positive_search_composite PASSED                                                                                                     [100%]

================================================================================= 1 passed in 61.55 seconds ====================================================================
```

File: tests/foreman/ui_airgun/test_contentview.py [hunk 6]
```diff
@@ -1350,3 +1362,498 @@ def test_positive_update_filter_affected_repos(session, module_org):
...
+        assert 'Promoted to {}'.format(lce.name) in result['Status']
+        # remove the content view version
+        session.contentview.remove_version(cv_name, VERSION)
+        assert not session.contentview.search_version(cv_name, VERSION)
+
+
+@skip_if_os('RHEL6')
+@upgrade
+@tier3
+def test_positive_rh_mixed_content_end_to_end(session):
```
Comment (ldjebran) [inline tests/foreman/ui_airgun/test_contentview.py]: missing @run_in_one_thread

**Pull request merged**